examples/flush-notes/src/services/notes.ts

```diff
@@ -0,0 +1,144 @@
+"use client";
+
+import { waku } from "@/services/waku";
+import { CONTENT_TOPIC } from "@/const";
+import {
+  symmetric,
+  generateSymmetricKey,
+} from "@waku/message-encryption/crypto";
+import {
+  createDecoder,
+  createEncoder,
+  Decoder,
+  Encoder,
+  IDecodedMessage,
+  Unsubscribe,
+  utf8ToBytes,
+  bytesToUtf8,
+} from "@waku/sdk";
+import { generateRandomString } from "@/utils";
+import { bytesToHex, hexToBytes } from "@waku/utils/bytes";
+
+type Note = {
+  id: string;
+  content: string;
+  iv: string;
+};
+
+type NoteResult = {
+  id: string;
+  password?: string;
+};
+
+export class Notes {
+  private decoder: Decoder;
+  private encoder: Encoder;
+  private messages: IDecodedMessage[] = [];
+  private subscription: undefined | Unsubscribe;
+
+  constructor() {
+    this.decoder = createDecoder(CONTENT_TOPIC);
+    this.encoder = createEncoder({ contentTopic: CONTENT_TOPIC });
+  }
+
+  public async createNote(
+    content: string,
+    toEncrypt?: boolean
+  ): Promise<NoteResult> {
+    const symmetricKey = toEncrypt ? generateSymmetricKey() : undefined;
+    const note = toEncrypt
+      ? await this.encryptNote(content, symmetricKey)
+      : { id: generateRandomString(), content, iv: undefined };
```

Comment:
It may be interesting to actually encrypt all notes by default.

Selective encryption should not be recommended as it is generally reduce anonymity properties by reducing the anonymity set.